notebooks/Notebook3.md

```diff
@@ -2,33 +2,66 @@
+I ended up just storing JSON in this file, but still opted for having it as a JS file so I don't have to use web requests to load it into a variable.
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+This week I wrote the parser for transitions and added a few features.
+For writing the transition parser, I ended up deciding to have a basic JavaScript file that has the functions all presentations need.
+Then separately, I can source the output of my transition parser which I decided to make a JSON list.
+Currently, I have it as a list with each element representing a transition and each transition containing a list the elements and effects to apply.
+I have all the transitions simplified into three types now: adding/removing classes, adding/removing transformations (translations, rotations, scaling, stretching), and changing element styles
+
+With both parsers working, I was able to replicate my presentation from last week into a [copy](<https://hmc-cs111-spring2023.github.io/project-code-occamkg/work%20files/testPres0/output>) produced by my program from the code.
+I also added a new [test presentation](<https://hmc-cs111-spring2023.github.io/project-code-occamkg/work%20files/testPres1/presentation>)!
+Some things I wanted to show with this are translating and scaling objects (images in this case), adding and removing classes (the blur added/removed from the text), and full size background images (the puppy!).
+The source input for this presentation can be found [here](<https://github.com/hmc-cs111-spring2023/project-code-occamkg/tree/master/work%20files/testPres1>) with the output in the `presentation` folder.
+I'm pretty satisfied with the transition syntax, but the element syntax is a bit bulky, so I may look into simplifying it further.
+I also added the slide number with some navigation controls, so that will be on every presentation.
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+Currently, whenever I want to render a change, I first have to save my file and then run the command `python path/to/program/run.py`.
+VS Code already has a very convenient feature (Open with Live Server) that rerenders HTML pages whenever you save anything in the workspace,
```

Comment:
I you want to make a change to the html, do you have to delete the previously generated html file and output a new one when you run the code, or does your program just make the changes to an already existing file.